src/coreclr/jit/optimizer.cpp

```diff
@@ -4457,7 +4449,15 @@ PhaseStatus Compiler::optUnrollLoops()
                 // or from the previous unroll clone of the bottom block (subsequent iterations).
                 // After doing this, all the newly cloned blocks now have proper flow and pred lists.
                 //
-                BasicBlock* const clonedTop = blockMap[loop.lpTop];
+                BasicBlock* const clonedTop     = blockMap[loop.lpTop];
+                BasicBlock*       clonedTopPrev = clonedTop->Prev();
+
+                if (clonedTopPrev->KindIs(BBJ_ALWAYS))
+                {
+                    assert(!clonedTopPrev->HasJump());
+                    clonedTopPrev->SetJumpDest(clonedTop);
+                }
```

Comment:
What other kind can `clonedTopPrev` be? It seems odd that for other cases we do not need to do any retargeting. Is it because the still remaining fallthrough logic of `BBJ_COND` covers it?

Reply:
> Is it because the still remaining fallthrough logic of BBJ_COND covers it?

Yes, it is still possible for `clonedTopPrev` to be a `BBJ_COND` and fall through into `clonedTop`. I'll add a comment to point this out.